ci/ciseedjobcheck.yaml

```diff
@@ -0,0 +1,210 @@
+                        }
+                        else
+                        {
+                            echo "Stage assertion failed"
+                        }
+                    }
+                    else
+                    {
+                        echo "Th stage $stage_number, should not be present in the pipeline, checking for it"
+                        build_status = sh(returnStdout: true, script: "oc get builds ${build_id} -o template  --template='{{index .metadata.annotations \"openshift.io/jenkins-status-json\"}}'").trim()
+                        def build_status_obj = new JsonSlurperClassic().parseText(build_status)
+                        if ( build_status_obj.stages[stage_number] == null ){
+                            echo "Stage not found as expected, assertion passed"
+                            actual_status = true
+                        }
+                        else
+                        {
+                            echo "Assertion failed"
+                        }
+                    }
+                    if (!actual_status)
+                    {
+                        currentBuild.result = 'FAILURE'
+                        error "Check failed: Going to fail state"
+                    }
+                }
+                def checkBuildConfigAvailability(expected_bc_name, expected_status)
+                {
+                    echo "build_config: ${expected_bc_name} expected_status: ${expected_status}"
+                    def actual_status = false
+
+                    bc_name = sh(returnStdout: true, script: "oc get bc ${expected_bc_name} -o name --ignore-not-found")
+
+                    //This function checks for a build config to be existing or not.
+                    if ( expected_status == "available" )
+                    {
+                        if (bc_name.contains(expected_bc_name))
+                        {
+                            echo "Build Config is existing assertion passed"
+                            actual_status = true
+                        }
+                        else
+                        {
+                            echo "BC availabilty assertion failed"
+                        }
+                    }
+                    else if(expected_status == "not_available")
+                    {
+                        if( bc_name.contains(expected_bc_name) )
+                        {
+                            echo "BC availability assertion failed"
+                        }
+                        else
+                        {
+                            echo "BC does not exists assertion passed"
+                            actual_status = true
+                        }
+                    }
+                    if (!actual_status)
+                    {
+                        currentBuild.result = 'FAILURE'
+                        error "Check failed: Going to fail state"
+                    }
+                }
```

Comment:
@bamachrn : having inline Jenkinsfile is not going to turn up well in future. Imagine the template size growing, it would be difficult to maintain. Could you please try with importing the Jenkinsfile here? and move all the groovy code to separate Jenkinsfile and re-try ?